### PR TITLE
ENG-11149 explainview vmc

### PR DESCRIPTION
**Pull request by yzhang1991**
This pull request addressed the following issues:
* Integrated Andrew's comments about the presentation of the table.
* Refactored the code for generating the `ExplainView` table. A new class `ViewExplainer` was created so that both `ReportMaker` and `ExplainView` system procedure and share the same code for table generation.
* Added a ReportMaker test case to verify the HTML table generated in VMC is correct.

### CODE REVIEW

File: src/frontend/org/voltdb/compilereport/ViewExplainer.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import org.voltdb.catalog.Table;
+import org.voltdb.types.ExpressionType;
+import org.voltdb.utils.CatalogUtil;
+import org.voltdb.utils.Encoder;
+
+// Generate a report that can explain to the users how we maintain a view.
+public class ViewExplainer {
+
+    private static String getIndexNameUsedInStatement(Statement stmt) {
+        // We should get at most one index from the statement.
```
Comment (wolffcm) [inline src/frontend/org/voltdb/compilereport/ViewExplainer.java]: For views with joins and/or where clauses, isn't it possible to have two indexes?

Comment (wolffcm) [inline src/frontend/org/voltdb/compilereport/ViewExplainer.java]: Oh, I think this is only for MIN/MAX refresh, not for initial materialization... still it seems possible to have more than one index used?

File: tests/frontend/org/voltdb/compilereport/TestReportMaker.java [hunk 2]
```diff
@@ -244,6 +244,32 @@ public void testTableWithoutIndexGetFullTableScanWarning() throws IOException {
         assertTrue(report.contains("Table " + tableName + " doesn't have any unique index, it will cause full table scans to update/delete DR record and may become slower as table grow."));
     }
 
+    public void testExplainViewTable() throws IOException {
```
Comment (wolffcm) [inline tests/frontend/org/voltdb/compilereport/TestReportMaker.java]: This seems like quite sufficient testing to me, for the changes you're making here.

File: src/frontend/org/voltdb/compilereport/ViewExplainer.java (hunk 1 above)
Comment (yzhang1991) [inline src/frontend/org/voltdb/compilereport/ViewExplainer.java]: This `getIndexNameUsedInStatement()` function is only used to extract the name of the index that was used in a statement in single-table view cases. For views with joins, we always directly output the plan explanation text.
The queries we used to refresh a MIN/MAX column in single-table view cases are always like this:
```sql
SELECT MIN(col) FROM T WHERE g1 = ? AND g2 = ? AND col >= ?;
```
I cannot think of a case where such queries can reference more than one index for evaluation. Do you think I missed anything?

Comment (wolffcm) [inline src/frontend/org/voltdb/compilereport/ViewExplainer.java]: If it's only for single-table view cases, I think it's fine.  You can only use one index per table instance in a statement.  Thanks for clarifying.  Maybe update the comment?

**Pull request merged**